CRM/Civiquickbooks/Contact.php

```diff
@@ -576,11 +576,15 @@ protected function getQBOContacts($start_date_string) {
    *                   to be an Organisation otherwise.
    */
   protected function getQBOContactByName($name, $givenName = NULL) {
-    $query = (
-      empty($givenName)
-      ? sprintf("SELECT * FROM Customer WHERE FullyQualifiedName = '%s'", addslashes($name))
-      : sprintf("SELECT * FROM Customer WHERE FamilyName = '%s' AND GivenName = '%s'", addslashes($name), addslashes($givenName))
-    );
+    $givenName = trim($givenName);
+    $inputParam = ['1' => [$name, 'String']];
+    $query = 'SELECT * FROM Customer WHERE FullyQualifiedName = %1';
+    if (!empty($givenName)) {
+      $inputParam['2'] = [$givenName, 'String'];
+      $query = 'SELECT * FROM Customer WHERE FamilyName = %1 AND GivenName = %2';
+    }
+    // Switch to CiviCRM to handle the single quote present in the contact name.
+    $query = CRM_Core_DAO::composeQuery($query, $inputParam);
```

Comment:
@sunilpawar what is the problem here you are trying to solve? Please provide examples

Reply:
Sometimes the name contains single or double quotes that are not properly escaped, causing QB to throw an error.

Reply:
I don't recall the exact reason for switching to the CiviCRM function, so I'll revert these changes for now.